### PR TITLE
feat: Add breadcrumbs for HTTP requests

### LINKED ISSUE #1197: HTTP Request Breadcrumbs
With https://github.com/getsentry/sentry-cocoa/pull/1178 we add auto performance instrumentation for HTTP requests. Having this in place it should be rather "easy" to add breadcrumbs for HTTP requests as this is an [expected feature](https://develop.sentry.dev/sdk/features/#http-client-integrations) for SDKs.

### PR DESCRIPTION
**Pull request by brustolin**
## :scroll: Description

Adding a breadcrumb in the scope for each HTTP request intercepted.

## :bulb: Motivation and Context

Closes #1197 

## :green_heart: How did you test it?

Unit tests

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [ ] I updated the docs if needed
- [ ] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

**Review by brustolin — COMMENTED** [review]
I'm wondering if we should create a breadcrumb in case the user cancels or suspend the request. It will be a breadcrumb with only URL and method without a status code.

Java add in data field the request body size and response body size, dart adds the request duration, the documentation does not mention either of this information. Should cocoa include any of this?

**Comment by philipphofmann** [comment]
> I'm wondering if we should create a breadcrumb in case the user cancels or suspend the request. It will be a breadcrumb with only URL and method without a status code.

The [docs](https://develop.sentry.dev/sdk/features/#http-client-integrations) state to only add a breadcrumb after the request finishes.

> Java add in data field the request body size and response body size, dart adds the request duration, the documentation does not mention either of this information. Should cocoa include any of this?

We have docs here @brustolin https://develop.sentry.dev/sdk/features/#http-client-integrations.

**Comment by philipphofmann** [comment]
I updated the docs for this https://github.com/getsentry/sentry-docs/pull/3954

File: Sources/Sentry/SentryNetworkTracker.m [hunk 1]
```diff
@@ -145,15 +146,29 @@ - (void)observeValueForKeyPath:(NSString *)keyPath
         return;
     }
 
+    SentryLevel breadcrumbLevel = sessionTask.error != nil ? kSentryLevelError : kSentryLevelInfo;
```
Comment (philipphofmann) [inline Sources/Sentry/SentryNetworkTracker.m:149]: `h`: I think we should add a breadcrumb even if the `netSpan` is `nil`.

**Review by philipphofmann — COMMENTED** [review]
`h`: We can add breadcrumbs even if there is no transaction in the scope, which we don't do currently.

**Review by philipphofmann — APPROVED** [review]
Thanks 🙏 . LGTM

**Pull request merged**